src/model/openthaigpt_pretraining_model/lightning/utils.py

```diff
@@ -96,7 +105,11 @@ def __init__(
         xformers: bool = False,
         checkpoint: bool = False,
         checkpoint_only_attention: bool = False,
+        num_nodes: int = 1,
     ):
+        if torch.cuda.get_device_name(0) == "NVIDIA A100-SXM4-40GB":
+            torch.set_float32_matmul_precision("medium")  # high
```

Comment:
Can you check is this really help with performance (Memory and Iteration per sec)

Reply:
It's warning when train with A100
```
You are using a CUDA device ('NVIDIA A100-SXM4-40GB') that has Tensor Cores.  
To properly utilize them, you should set 
`torch.set_float32_matmul_precision('medium' | 'high')`  
which will trade-off precision for performance. 
For more details, read https://pytorch.org/docs/stable/generated/torch.set_float32_matmul_precision.html#torch.set_float32_matmul_precision
```
result when change from lightning (Tensor Cores and Matrix Multiplication Precision) [ref](https://lightning.ai/pages/community/tutorial/accelerating-large-language-models-with-mixed-precision-techniques/)
![image](https://github.com/OpenThaiGPT/openthaigpt-pretraining/assets/101176694/1dc3cc50-a3d8-4edb-9e8e-e62e79c20307)